src/policy/mod.rs

```diff
@@ -143,17 +144,16 @@ pub struct BasePolicy {
     /// SleepBetweenReplies determines duration to sleep between retries if a
     /// transaction fails and the timeout was not exceeded.  Enter zero to skip sleep.
     pub sleep_between_retries: Option<Duration>,
+
+    /// Optional FilterExpression
+    pub filter_expression: Option<FilterExpression>,
```

Comment:
I think we should also add

    fn filter_expression(&self) -> &Option<FilterExpression> 

to the `Policy` trait, and implement it for

    impl<T> Policy for T where T: PolicyLike

All policies that include a base policy and implement `PolicyLike` would then automatically support filter expressions.

Reply:
I tried that. I cant define the function with the PolicyLike implement, because the variable does not sit at base_policy level. Putting it there would force me to save mut BasePolicy instead of normal BasePolicy. 
That will force us to actively set lifetimes and causes multiple problems around Size/Clone etc.
Would be more clean, but is not possible without big changes.

Reply:
> But then why add this field to the BasePolicy struct at all?

Because BasePolicy is ReadPolicy. Thats the only reason.

> I'm also not quite sure I understand why adding the filter expressions to the base policy would require changing the mutability of the base policy? How would that be different from e.g.

i actually never tried it like that. But good point. I had in mind that both need to be mut. I will try later today.

Reply:
> Because BasePolicy is ReadPolicy. Thats the only reason.

Ah, right. Forgot about that. That was probably a design decision that should be revisited at some point...

Reply:
I wanted it to be on first level. Like wpolicy.filter_expression. That would require a set_filter method and for that it needs to be mut

Reply:
Ok, let's keep it the way you have implemented it. I also don't really like that users have to know which settings are part of the policy and which are part of the base policy, e.g. `wpolicy.send_key` vs `wpolicy.base_policy.timeout`. Especially given how the read policy does not follow this same pattern. It's a bit of a mess, and probably should be revisited at some point. But I think for now we're good.